submissions/Chikus/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,69 @@
+const Net = require('net');
+var host = '';
+var lower_port = 0;
+var higher_port = 65535;
+var port_list = [];
+var timeout = 300;
+
+function check(host, port, callback) {
+ var socket = Net.createConnection(port, host);
+ var timer  = setTimeout(function () {
+  socket.destroy();
+  callback(false);
+}, timeout);
+ socket.once('connect', function () {
+  clearTimeout(timer);
+  socket.destroy();
+  process.stdout.write('.');
+  port_list.push(port);
+  callback(true);
+ });
+ socket.on('error', function () {
+  clearTimeout(timer);
+  callback(false);
+ });
+}
+
+function run_ports(){
+  check(host,lower_port, function next_check(result) {
+    if (lower_port == higher_port) {
+      if (port_list.length) {
+          process.stdout.write('\n'+port_list.join()+' ports are opened \n');
+      }
+      else {
+        process.stdout.write('No ports were found \n');
+      }
+    }
+    else {
+      check(host,++lower_port,next_check);
```

Comment:
Double self-referencing (check, next_check) is not obvious and a bit dangerous code.

Reply:
Ok I will change this, but what you are saying is that, recursion with the same function is not a good practice even if I use a call back, I think after I destroy the socket there is nothing behind I start with the new function. but no problem consider this done. Thanks :D

Reply:
I'm not saying you have leak here, I'm just emphasising that it's easier to make mistake in such code.